src/file_storage/file_storage.ts

```diff
@@ -1,20 +1,40 @@
 import Result from "../result"
 import {Readable} from "stream";
 import Config from "../config"
-import MockFileStorageImpl from "./mock_impl";
 import AzureBlobImpl from "./azure_blob_impl";
+import MockFileStorageImpl from "./mock_impl";
+import Filesystem from "./filesystem_impl";
 
 export interface FileStorage {
-    uploadPipe(filepath:string, stream: Readable | null, encoding?:string, mimeType?:string): Promise<Result<string>>
+    uploadPipe(filepath:string, filename: string, stream: Readable | null, contentType?:string, encoding?:string,): Promise<Result<FileUploadResponse>>
     deleteFile(filepath:string): Promise<Result<boolean>>
+    name(): string
 }
 
-// Returns an instantiated FileStorage provider if provider method is set.
-export default function FileStorageProvider(): FileStorage | null {
-    switch (Config.file_storage_method) {
+export type FileUploadResponse = {
+    filename: string
+    filepath: string
+    size: number // size in KB
+    metadata: {[key:string]: any} // adapter specific metadata if needed
+    adapter_name: string
+}
+
+// Returns an instantiated FileStorage provider if provider method is set, or user
+// can provide a specific adapter name to fetch an instance of specified adapter
+export default function FileStorageProvider(adapterName?: string): FileStorage | null {
+    switch ((adapterName) ? adapterName : Config.file_storage_method) {
         case "azure_blob_storage": {
```

Comment:
I'm not sure what the calls to this function look like, but the switch for the `adapterName` parameter has a case for "azure_blob_storage", but the actual implementation for the azure_blob file storage returns a name of "azure_blob". Is there a discrepancy there?

Reply:
Good catch, I'll correct that